python/pandas-eval.yaml

```diff
@@ -0,0 +1,48 @@
+rules:
+  - id: pandas-eval
+    message: >-
+      Pandas eval() and query() may be dangerous if used to evaluate
+      dynamic content. If this content can be input from outside the program, this
+      may be a code injection vulnerability. Ensure evaluated content is not definable
+      by external sources.
+    languages: [python]
+    severity: ERROR
+    metadata:
+      category: security
+      cwe: "CWE-95: Improper Neutralization of Directives in Dynamically Evaluated Code ('Eval Injection')"
+      subcategory:
+        - audit
+      confidence: LOW
+      likelihood: LOW
+      impact: HIGH
+      technology: [pandas]
+      description: "Potential arbitrary code execution from `pandas` functions that evaluate user-provided expressions"
+      references:
+        - https://blog.trailofbits.com/2021/03/15/never-a-dill-moment-exploiting-machine-learning-pickle-files/
+
+    patterns:
+      - pattern-inside: |
+          import pandas
+          ...
+      - pattern-either:
+          - patterns:
+              - pattern: pandas.DataFrame.$FN(...)
+              - pattern-not: pandas.DataFrame.$FN("...", ...)
+              - pattern-not: pandas.DataFrame.$FN(f"", ...)
```

Comment:
Is `f""` intended? We're trying to remove only empty f-strings?

Reply:
It is intended. We want to remove all static strings, including empty strings. Semgrep seems to have a bug - according to semgrep:
* `f" expr "` - is static string
* `f" {expr} "` - is dynamic string
* `f""` - is dynamic string

Will add a testcase for that and make issue in the upstream r2c

Reply:
https://github.com/semgrep/semgrep/issues/10047